src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -712,6 +712,10 @@ static inline void populate_event(reservation_t& res,
     metadata_from_attributes(res, obj);
   }
   event.x_meta_map = res.x_meta_map;
+  if (0 == event.object_size) {
+    event.object_size = obj->get_obj_size();
```

Comment:
did you test to see that this is fixing the issue?
from looking at: https://github.com/ceph/ceph/blob/7179c0e0c73ba745e01b82b19f36ef75f7edde69/src/rgw/driver/rados/rgw_cr_rados.cc#L816
the value of `event.object_size` should be set from `dest_obj.get_obj_size()` which is the same as `obj->get_obj_size()`.

I think that in this case we should pass the `bytes_transferred` to `rgw::notify::publish_commit()` in `rgw_cr_rados.cc`

Reply:
i think the fix works because fetching the metadata also populate the size.
but this is not guaranteed to happen (if metadata was fetched before).

Reply:
Thought about this and did some tests. In my tests, it seems the only occasion of object size being changed is when the object is overwritten, in which case the `metadata_fetched_from_attributes` is false. Or did I miss something here?
I can try and explorer passing `bytes_transferred` though.

Reply:
yes. if `metadata_fetched_from_attributes` is `false` the object metadata is fetched, and the size myab get the real value.
but, this API can be called from different places, it looks risky that thew line the refetched the object size is relying on the metadata fetching code

Reply:
Pushed the changes to use `bytes_transferred ` and also populate the owner.
Did the tests following https://gist.github.com/yuvalif/527a5064a1748d447129c594838448b0.
All look good except the metadata. 
In metadata section in the following example output, sync notification is missing metadata `x-amz-content-sha256`, `x-amz-date`, and `x-amz-tagging`. Is this intended or a bug?

-------------------------------------------------------
Regular notification:
_{
  "Records": [
    {
      "eventVersion": "2.2",
      "eventSource": "ceph:s3",
      "awsRegion": "zg1",
      "eventTime": "2023-05-01T16:17:34.844165Z",
      "eventName": "ObjectCreated:Put",
      "userIdentity": {
        "principalId": "zone.user"
      },
      "requestParameters": {
        "sourceIPAddress": ""
      },
      "responseElements": {
        "x-amz-request-id": "7afba4d2-3e1a-4ba6-a3a4-803ac89e5ba5.4158.11183591382816850133",
        "x-amz-id-2": "103e-zg1-1-zg1"
      },
      "s3": {
        "s3SchemaVersion": "1.0",
        "configurationId": "notif1",
        "bucket": {
          "name": "fish",
          "ownerIdentity": {
            "principalId": "zone.user"
          },
          "arn": "arn:aws:s3:zg1::fish",
          "id": "7afba4d2-3e1a-4ba6-a3a4-803ac89e5ba5.4160.1"
        },
        "object": {
          "key": "myfile",
          "size": 52428800,
          "eTag": "64265add918ea07b8a4c1863bd44715d",
          "versionId": "4OBdctjOTuVsM4M5.ySPUpdPcUmLCS1",
          "sequencer": "1EE64F6405208836",
          "metadata": [
            {
              "key": "x-amz-content-sha256",
              "val": "a347f7cb986faa748b474f809d21b6d7369897fa39c5a63ba50db2c701ad2121"
            },
            {
              "key": "x-amz-date",
              "val": "20230501T161734Z"
            },
            {
              "key": "x-amz-meta-foo",
              "val": "bar"
            },
            {
              "key": "x-amz-meta-ka",
              "val": "boom"
            },
            {
              "key": "x-amz-tagging",
              "val": "Key1=Value1"
            }
          ],
          "tags": [
            {
              "key": "Key1",
              "val": "Value1"
            }
          ]
        }
      },
      "eventId": "1682957854.914890.64265add918ea07b8a4c1863bd44715d",
      "opaqueData": ""
    }
  ]
}_

-------------------------------------------------------
Sync notification:
_{
  "Records": [
    {
      "eventVersion": "2.2",
      "eventSource": "ceph:s3",
      "awsRegion": "zg1",
      "eventTime": "2023-05-01T16:17:52.281273Z",
      "eventName": "ObjectSynced:Create",
      "userIdentity": {
        "principalId": "rgw sync"
      },
      "requestParameters": {
        "sourceIPAddress": ""
      },
      "responseElements": {
        "x-amz-request-id": "0",
        "x-amz-id-2": "102a-zg1-2-zg1"
      },
      "s3": {
        "s3SchemaVersion": "1.0",
        "configurationId": "notif1",
        "bucket": {
          "name": "fish",
          "ownerIdentity": {
            "principalId": "zone.user"
          },
          "arn": "arn:aws:s3:zg1::fish",
          "id": "7afba4d2-3e1a-4ba6-a3a4-803ac89e5ba5.4160.1"
        },
        "object": {
          "key": "myfile",
          "size": 52428800,
          "eTag": "64265add918ea07b8a4c1863bd44715d",
          "versionId": "4OBdctjOTuVsM4M5.ySPUpdPcUmLCS1",
          "sequencer": "30E64F644505C410",
          "metadata": [
            {
              "key": "x-amz-meta-foo",
              "val": "bar"
            },
            {
              "key": "x-amz-meta-ka",
              "val": "boom"
            }
          ],
          "tags": [
            {
              "key": "Key1",
              "val": "Value1"
            }
          ]
        }
      },
      "eventId": "1682957872.281281.64265add918ea07b8a4c1863bd44715d",
      "opaqueData": ""
    }
  ]
}_

Reply:
it is actually a bug that these attributes are in the regular notification. the definition of metadata attributes are ones that start with `x-amz-meta...`
so, the sync notification is correct, but the regular notification added an attributes that should not be there (there are usually added by the client code and not the user)

Reply:
Got it. Thanks for the explanation. I'll put in a separate PR to remove them from the regular notification.

Reply:
this is great, but please open a tracker as well.
we will probably need to backport that to quincy and pacific

Reply:
sure. will do

Reply:
opened https://tracker.ceph.com/issues/59592 to remove non `x-amz-meta...` attributes from the `metadata` section of the regular notifications.